### PR TITLE
Nether wart growing

### LINKED ISSUE #1549: Can't plant nether wart
Can not plant nether wart. It just doesn't appear on the soul sand block when player taps this block with the wart in his hand.

### PR DESCRIPTION
**Pull request by Leonidius20**
Now you can plant nether wart and it will grow. Fixes #1549 

P.S. I'm sorry for my old commits here, I updated my master branch from upstem and made this pull request from a new branch, but it didn't help

screenshot_2017-03-22-14-20-05

### CODE REVIEW

**Comment by Tee7even** [comment]
>P.S. I'm sorry for my old commits here, I updated my master branch from upstem and made this pull request from a new branch, but it didn't help

Refork the repo.

**Comment by Leonidius20** [comment]
If I'll refork the repo now, this PR will be deleted? Or no?

**Comment by Tee7even** [comment]
It will get closed, so wait.

File: src/main/java/cn/nukkit/block/BlockNetherWart.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    @Override
+    public int onUpdate(int type) {
+        if (type == Level.BLOCK_UPDATE_NORMAL) {
+            if (this.getSide(Vector3.SIDE_DOWN).getId() != SOUL_SAND) {
+                this.getLevel().useBreakOn(this);
+                return Level.BLOCK_UPDATE_NORMAL;
+            }
+        } else if (type == Level.BLOCK_UPDATE_RANDOM) {
+            if (new Random().nextInt(10) == 1) {
+                if (this.meta < 0x02) {
```
Comment (Tee7even) [inline src/main/java/cn/nukkit/block/BlockNetherWart.java]: Grows only to second stage?

File: src/main/java/cn/nukkit/block/BlockNetherWart.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    BlockNetherWart block = (BlockNetherWart) this.clone();
```
Comment (Tee7even) [inline src/main/java/cn/nukkit/block/BlockNetherWart.java]: Unnecesary.

File: src/main/java/cn/nukkit/block/BlockNetherWart.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Override
+    public int onUpdate(int type) {
+        if (type == Level.BLOCK_UPDATE_NORMAL) {
+            if (this.getSide(Vector3.SIDE_DOWN).getId() != SOUL_SAND) {
+                this.getLevel().useBreakOn(this);
+                return Level.BLOCK_UPDATE_NORMAL;
+            }
+        } else if (type == Level.BLOCK_UPDATE_RANDOM) {
+            if (new Random().nextInt(10) == 1) {
```
Comment (Leonidius20) [inline src/main/java/cn/nukkit/block/BlockNetherWart.java]: It has 10% chance to grow up if the block updates.

File: src/main/java/cn/nukkit/block/BlockNetherWart.java (hunk 1 above)
Comment (Leonidius20) [inline src/main/java/cn/nukkit/block/BlockNetherWart.java]: The third (the last) stage has 0x02 meta

File: src/main/java/cn/nukkit/block/BlockNetherWart.java (hunk 2 above)
Comment (Leonidius20) [inline src/main/java/cn/nukkit/block/BlockNetherWart.java]: It is used to send the old block state and the new block state to BlockGrowEvent

**Comment by Leonidius20** [comment]
What else I need to change?

File: src/main/java/cn/nukkit/block/BlockNetherWart.java (hunk 1 above)
Comment (Tee7even) [inline src/main/java/cn/nukkit/block/BlockNetherWart.java]: It has 4 stages, 2 in between have the same appearance.

**Comment by Leonidius20** [comment]
That was my mistake with ID's. Now it grows correctly in 4 states.

File: src/main/java/cn/nukkit/block/BlockNetherWart.java [hunk 4]
```diff
@@ -84,7 +84,7 @@ public int getId() {
 
     @Override
     public int[][] getDrops(Item item) {
-        if (this.meta >= 0x02) {
+        if (this.meta >= 0x03) {
```
Comment (Tee7even) [inline src/main/java/cn/nukkit/block/BlockNetherWart.java]: Change this to `==` and I'll merge it.

**Comment by Leonidius20** [comment]
@Tee7even ready.

**Pull request merged**